checkPropTypes.js

```diff
@@ -7,16 +7,21 @@
 
 'use strict';
 
+var usage = require('./usage');
+
 var printWarning = function() {};
 
-if (process.env.NODE_ENV !== 'production') {
+if (usage.shouldDoSomething()) {
```

Comment:
this change will prevent bundling optimizations they delete this chunk of code in production.